### PR TITLE
PMM-11651: fix full access role bug

### PR DESCRIPTION
**Pull request by michal-kralik**
When full access role is combined with limited role, full access was not granted

PMM-11651

Build: SUBMODULES-3330

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

**Review by recharte — APPROVED** [review]

File: managed/services/grafana/auth_server.go [hunk 1]
```diff
@@ -356,9 +356,14 @@ func (s *AuthServer) getFiltersForVMProxy(userID int) ([]string, error) {
 
 	filters := make([]string, 0, len(roles))
 	for _, r := range roles {
-		if r.Filter != "" {
-			filters = append(filters, r.Filter)
+		if r.Filter == "" {
+			// Special case when a user has assigned a role with no filters.
+			// In this case it's irrelevant what other roles the user has assigned.
+			// The user shall have full access.
+			return nil, nil
```
Comment (atymchuk) [inline managed/services/grafana/auth_server.go:363]:
```suggestion
			return nil, nil //nolint:nilnil
```
We have a rule now :)

File: managed/services/grafana/auth_server.go [hunk 2]
```diff
@@ -356,9 +356,14 @@ func (s *AuthServer) getFiltersForVMProxy(userID int) ([]string, error) {
 
 	filters := make([]string, 0, len(roles))
 	for _, r := range roles {
-		if r.Filter != "" {
-			filters = append(filters, r.Filter)
+		if r.Filter == "" {
+			// Special case when a user has assigned a role with no filters.
+			// In this case it's irrelevant what other roles the user has assigned.
```
Comment (atymchuk) [inline managed/services/grafana/auth_server.go:361]:
```suggestion
			// In this case it's irrelevant what other roles are assigned to the user.
```

File: managed/services/grafana/auth_server.go (hunk 1 above)
Comment (michal-kralik) [reply managed/services/grafana/auth_server.go]: I changed it to return empty slice.
I'm wondering why nothing complained about the `nilnil` rule.

Comment (atymchuk) [reply managed/services/grafana/auth_server.go]: Because you haven't rebased after I merged that rule, no?

**Review by atymchuk — APPROVED** [review]

**Pull request merged**